### PR TITLE
add learning materials tab

### PR DESCRIPTION
**Pull request by ChristopherChudzicki**
### What are the relevant tickets?
Closes https://github.com/mitodl/hq/issues/4620

### Description (What does it do?)
This PR adds a "Learning Materials" tab to the search page and channel pages

### Screenshots (if ap
Screenshot 2024-06-20 at 4 47 03 PM
propriate):
Screenshot 2024-06-20 at 4 35 03 PM
Screenshot 2024-06-20 at 4 35 18 PM
Screenshot 2024-06-20 at 4 35 37 PM

### How can this be tested?
On this branch:
1. View the search page locally. 
2. You should see 4 tabs: "All", "Courses", "Programs", "Learning Materials".
    - Check that each tab shows the appropriate things
    - "Learning Materials" should show everything except courses and programs
3. Check that tab counts do not change when switching tabs

### CODE REVIEW

File: frontends/mit-open/src/page-components/SearchDisplay/SearchDisplay.tsx [hunk 1]
```diff
@@ -474,8 +482,24 @@ interface SearchDisplayProps {
...
+   *
+   * This is partially addressed by `@mitodl/course-search-utils`, which exports
+   * a wrapper around `useSearchParams`: subsequent calls to `setSearchParams`
+   * DO use the current value, with one caveat: The setSearchParams function
+   * must be from the same "instance" of `useSearchParams`.
+   *
+   * Because of this, we pass the setSearchParams function from the parent
+   * rather than from a new "instance" of `useSearchParams`.
+   */
+  setSearchParams: UseResourceSearchParamsProps["setSearchParams"]
```
Comment (ChristopherChudzicki) [inline frontends/mit-open/src/page-components/SearchDisplay/SearchDisplay.tsx:502]: I opened an issue about this in course-search-utils https://github.com/mitodl/course-search-utils/issues/113 with more details.

**Comment by Ferdi** [comment]
@ChristopherChudzicki in the main nav menu we have "learning materials" as tbd. Are you planning to remove tbd and link to this tab?

**Comment by ChristopherChudzicki** [comment]
> @ChristopherChudzicki in the main nav menu we have "learning materials" as tbd. Are you planning to remove tbd and link to this tab?

Good call. I'll add that.

File: frontends/mit-open/src/page-components/SearchDisplay/SearchDisplay.tsx [hunk 2]
```diff
@@ -489,19 +513,32 @@ const SearchDisplay: React.FC<SearchDisplayProps> = ({
   setParamValue,
   clearAllFacets,
   toggleParamValue,
-  patchParams,
   showProfessionalToggle,
+  setSearchParams,
 }) => {
+  const [searchParams] = useSearchParams()
+  const activeTab =
+    TABS.find((t) => t.name === searchParams.get("tab")) ??
```
Comment (ChristopherChudzicki) [inline frontends/mit-open/src/page-components/SearchDisplay/SearchDisplay.tsx:521]: Previously the active tab was based on `resource_type` query parameter.

Now the tabs are:
- All `tab="all"` ... `resource_type=[]`
- Courses `tab="courses"` ... `resource_type=["course"]`
- Programs `tab="programs"` ... `resource_type=["program"]`
- Learning Materials `tab="learning-materials"` ... `resource_type=["video", "podcast", "podcast_episode", "video_playlist", "learning_path"]`.

I was a little torn about whether to infer the tab based on `resource_type` query param, or use a separate `tab` param (which is what I ended up doing).

Using a separate tab really simplifies URLs like `href: querifiedSearchUrl({ tab: "learning-materials" }),` in the navbar. 

My thought is that **in the future** when we add a "resource_type" sidebar facet (for the LM tab), the URLs would look like `?tab=learning-materials&resource_type="video"`.

**Pull request merged**